sql/wsrep_sst.cc

```diff
@@ -27,6 +27,75 @@
 #include "wsrep_xid.h"
 #include <cstdio>
 #include <cstdlib>
+#include <functional>
+#include <sstream>
+#include <string>
+
+/* We don't need PSI instrumentation for regex mutex, however
+   Regex class has its constructor compiled conditionally.
+   Disable instrumentation. 
+   Regex class uses c++11 nullptr. As we are in c++03 world
+   let's define nullptr to satisfy compiler */
+#ifdef HAVE_PSI_INTERFACE
+#undef HAVE_PSI_INTERFACE
+#endif
+#define nullptr 0
+#include <sql_regex.h>
+
+/* Check if the string is valid. In method name we allow only:
+   alpha-num
+   underscore
+   dash
+   but w do not allow leading and trailing dash.
```

Comment:
It is probably better to say `\w` instead of just `w`.
```suggestion
   but \w does not allow leading or trailing dashes.
```